### ISSUE TITLE
[BUG] Feed List reloads to top when switching between tabs

### ISSUE THREAD

**Issue by boscojwho** [Athlon007/Leomard#34]
**Describe the bug**
Scroll down main Feed list, then switch to another tab (e.g. Inbox), then switch back to Feed. Feed will do a fresh reload and start from the top.

**To Reproduce**
1. Load main Feed and scroll down a few pages.
2. Switch to another tab (e.g. Inbox).
3. Switch back to Feed.
4. Feed reloads to top.

**Expected behavior**
- Feed should stay where it was without loading.

**Information:**
- Version 0.1 (1)

**Comment by Athlon007** [comment]
Is there even a way to keep the position of a Feed, when switching between pages, without having it in memory at all time?

**Comment by boscojwho** [comment]
Not really, some parts of it will have to remain in memory, but this isn't an uncommon pattern (keeping multiple pages in memory).

But while thinking about this problem, I started wondering whether it might make more sense if `Feed`, `Inbox`, and `Search` should live in a system `TabView`, instead of being in the sidebar? Apollo did this on iPhone, and many other Lemmy apps also use the same pattern. Using a tab view would have the added benefit of not having these three pages reload when switching between them.

**Comment by boscojwho** [comment]
> Is there even a way to keep the position of a Feed, when switching between pages, without having it in memory at all time?

Not really, some parts of it will have to remain in memory, but this isn't an uncommon pattern (keeping multiple pages in memory).

But while thinking about this problem, I started wondering whether it might make more sense if `Feed`, `Inbox`, and `Search` should live in a system `TabView`, instead of being in the sidebar? Apollo did this on iPhone, and many other Lemmy apps also use the same pattern. Using a tab view would have the added benefit of not having these three pages reload when switching between them.

**Comment by Athlon007** [comment]
Alright, but I want to do it after the release of 0.2 (probably this Monday/Tuesday). I want to release 0.2, see if people encounter bugs, and release potential 0.2.1.

I will have limited time to work on Leomard from 29th July to 12th August, and ESPECIALLY I won't be able to work on it between 4th and 10th :)

**Comment by boscojwho** [comment]
Woohoo, 0.2 lets go =P
